streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -172,7 +172,7 @@
                                         REPLICATION_FACTOR_DOC)
                                 .define(TIMESTAMP_EXTRACTOR_CLASS_CONFIG,
                                         Type.CLASS,
-                                        ConsumerRecordTimestampExtractor.class.getName(),
+                                        FailingConsumerRecordTimestampExtractor.class.getName(),
```

Comment:
We should consider what renaming `ConsumerRecordTimestampExtractor` to `FailingConsumerRecordTimestampExtractor` means for backwards compatibility.